### PR TITLE
Update github workflow with CY tests

### PR DESCRIPTION
**Pull request by kelanik8**
# The following changes are implemented
TODO: Summary

# Changes in the user interface:
TODO: Add screenshots, recordings or remove this section

# Checklist when submitting a final (!draft) PR
 - [ ] Commits are tidied up, squashed if needed and follow guidelines in CONTRIBUTING.md
 - [ ] Code builds
 - [ ] All existing tests pass
 - [ ] All new critical code is covered by tests
 - [ ] PR is linked to the relevant issue(s)
 - [ ] Rebased with the target branch

### CODE REVIEW

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+# These jobs can be tested with nektos/act tool
+# https://github.com/nektos/act
+#
+# Look for "!env.ACT" in the DRY_RUN expression below
+
+on:
+  push:
+    branches:
+      - 2023.9.x-feature-e2e
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: change to main branch

File: .github/workflows/test-e2e.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+env:
+  BUILD_OS: linux
+  BUILD_ARCH: amd64
+
+  GO_VERSION: 1.19
+  GOFLAGS: -mod=readonly
+
+  NODE_VERSION: 16
+  CORTEZA_VERSION: 2023.9.0
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: do we still need this?

File: .github/workflows/test-e2e.yml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        client:
+          - { port: 8081, name: compose }
+          - { port: 8080, name: admin }
+          - { port: 8082, name: workflow }
+          - { port: 8083, name: reporter }
+          - { port: 8086, name: one }
+          # - { port: 8087, name: privacy }
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup tmate session
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: remove, this was for debugging purposes only

File: .github/workflows/test-e2e.yml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+          echo $SCREENSHOTS_GEN_FOLDER
+
+      - name: Create crust key
+        env:
+          RELEASE_CRUST_SFTP_KEY: ${{ secrets.RELEASE_CRUST_SFTP_KEY }}
+          RELEASE_CRUST_KEY_FILE: ${{ env.TMP_KEY }}
+        run: |
+          echo ${RELEASE_CRUST_SFTP_KEY} | base64 -d > ${RELEASE_CRUST_KEY_FILE} && \
+          echo ${RELEASE_CRUST_SFTP_KEY} > /tmp/foo
+      - name: Make folder for uploaded screenshots
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml:69]: newline before this line please

File: .github/workflows/test-e2e.yml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+      - name: "Setup YARN"
+        run: npm install -g yarn @vue/cli-service
+
+      - name: Checkout corteza repo
+        uses: actions/checkout@v3
+        with:
+          path: corteza
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: 2023.9.x-feature-e2e
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: you do not need to checkout this repo as you are already in it, remove this step, but check if the folders are ok

File: .github/workflows/test-e2e.yml [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        run: |
+          docker-compose up -d backend
+          docker-compose up -d postgres
+
+      - name: Update config file in ${{ matrix.client.name }} webapp
+        working-directory: corteza
+        env:
+          CLIENT_NAME: ${{ matrix.client.name }}
+        run: |
+          cp .github/workflows/assets/client/config.js client/web/${CLIENT_NAME}/public/config.js
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: :ok_hand:

File: .github/workflows/test-e2e.yml [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+            --use-aliases \
+            client \
+            sh -c "cd /corteza/lib/js && yarn link && cd /corteza/lib/vue && yarn link && cd /corteza/client/web/${CLIENT_NAME} && yarn link @cortezaproject/corteza-vue && yarn link @cortezaproject/corteza-js && yarn install && yarn serve --port ${CLIENT_PORT}"
+
+      - name: Clone CY repo
+        uses: actions/checkout@v3
+        with:
+          repository: cortezaproject/corteza-e2e-cypress
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: 2023.9.x-feature-e2e
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: checkout the main branch, set it as env on the top of this file

File: .github/workflows/test-e2e.yml [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+          path: corteza-e2e-cypress
+
+      # github actions do not support docker-compose --wait
+      - name: Wait for postgres healthcheck
+        run: timeout 180s sh -c 'until docker ps | grep corteza_postgres_1 | grep -q healthy; do echo "Waiting for container to be healthy..."; sleep 1; done'
+
+      - name: Wait for "${{ matrix.client.name }}" healthcheck
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: remove the `"`

File: .github/workflows/test-e2e.yml [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+      - name: Wait for backend healthcheck
+        run: timeout 180s sh -c 'until docker ps | grep corteza_backend_1 | grep -q healthy; do echo "Waiting for container to be healthy..."; sleep 1; done'
+
+      - name: Run CY tests for ${{ matrix.client.name }}
+        env:
+          CLIENT_NAME: ${{ matrix.client.name }}
+          CLIENT_PORT: ${{ matrix.client.port }}
+        working-directory: corteza-e2e-cypress
+        run: |
+          docker ps
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: remove, this is debugging statement

File: docker-compose.yaml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      retries: 5
+
+  client:
+    network_mode: host
+    image: node:16.20-alpine3.16
+    volumes:
+      - "${PWD}:/corteza"
+    working_dir: /corteza
+    environment:
+      BUILD_VERSION: 2023.3.6-3-g251c05abf
```
Comment (petergrlica) [inline docker-compose.yaml]: do we need this? if yes, why is it fixed?

File: docker-compose.yaml [hunk 11]
```diff
@@ -0,0 +1,60 @@
+version: "3.5"
```
Comment (petergrlica) [inline docker-compose.yaml]: This file should not be on the root of the repo, as it is github-related only.
Copy it to some other location (`.github/workflows/assets/docker-compose.yaml` maybe?) and use it in the github actions either directly (`docker-compose -f `) or copy before use.

File: docker-compose.yaml [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+      - "${PWD}:/corteza"
+    entrypoint: |
+      sh -c "
+        git config --global --add safe.directory /corteza &&
+        go run /corteza/server/cmd/corteza/main.go serve-api
+      "
+    working_dir: /corteza/server
+    environment:
+      - DB_DSN=postgres://corteza:root@localhost:5432/corteza_cy_test?sslmode=disable
+      # - GIN_ARG_LADDR=0.0.0.0
```
Comment (petergrlica) [inline docker-compose.yaml]: remove

**Review by petergrlica — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 10 above)
Comment (kelanik8) [reply docker-compose.yaml]: It's a requirement in `vue.config-builder.js` in each of the clients

File: .github/workflows/test-e2e.yml [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+            client \
+            sh -c "ls && cd /corteza/lib/js && yarn link && \
+                  cd /corteza/lib/vue && yarn link && \
+                  cd /corteza/client/web/${CLIENT_NAME} && \
+                  yarn link @cortezaproject/corteza-vue && \
+                  yarn link @cortezaproject/corteza-js && \
+                  yarn install && \
+                  yarn serve --port ${CLIENT_PORT}"
+          
+          docker logs ${CLIENT_NAME}
```
Comment (petergrlica) [inline .github/workflows/test-e2e.yml]: remove

**Review by petergrlica — APPROVED** [review]

**Pull request merged**